live-ingester/src/test/java/org/alfresco/hxi_connector/live_ingester/storage/StorageClientIntegrationTest.java

```diff
@@ -0,0 +1,113 @@
+ * This file is part of the Alfresco software.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail.  Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.hxi_connector.live_ingester.storage;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.testcontainers.containers.localstack.LocalStackContainer.Service.S3;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.List;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.DynamicPropertyRegistry;
+import org.springframework.test.context.DynamicPropertySource;
+import org.testcontainers.containers.localstack.LocalStackContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.utility.DockerImageName;
+
+import org.alfresco.hxi_connector.live_ingester.storage.local.LocalStorageClient;
+import org.alfresco.hxi_connector.live_ingester.storage.local.LocalStorageConfig;
+import org.alfresco.hxi_connector.live_ingester.utils.DockerTags;
+
+@SpringBootTest(classes = {LocalStorageConfig.class, SignedStorageClient.class})
+@ActiveProfiles({"test"})
+@Testcontainers
+class StorageClientIntegrationTest
+{
+    private static final String LOCALSTACK_IMAGE = "localstack/localstack";
+    private static final String LOCALSTACK_TAG = DockerTags.getOrDefault("localstack.tag", "3.0.2");
+    private static final String BUCKET_NAME = "test-hxinsight-bucket";
+    private static final String OBJECT_KEY = "dummy.txt";
+    private static final String OBJECT_CONTENT = "dummy's file dummy content";
+    private static final String OBJECT_CONTENT_TYPE = "plain/text";
+
+    @Container
+    @SuppressWarnings("PMD.FieldNamingConventions")
+    private static final LocalStackContainer localStack = new LocalStackContainer(DockerImageName.parse(LOCALSTACK_IMAGE).withTag(LOCALSTACK_TAG));
```

Comment:
Why didn't you want to call this `LOCAL_STACK`? Maybe you could add a comment to explain the suppression if this has to match with somewhere else in the code?

Reply:
Yeah, so there is a live debate in Java society about what should be written using SNAKE_CASE an what not. I believe that only constants should be written this way. What is a constant is very debatable, there is quite cool explanation here: https://stackoverflow.com/a/12069687/2211974 - I like it 😅 Overall IMO `localStack` is a final reference not a CONSTANT, so I called it using camelCase.

Reply:
Ah - yes, I read that before. However since we've all had years of seeing `CAPS == static final` then the discussion seemed a bit moot. I think the convenience of using the `@Slf4j` annotation is worth an exception however. Overall I don't mind either way, but if we want to use this style more widely we should probably remove the PMD check.